### ISSUE TITLE
FreeBSD - getAddrInfo does not exist

### ISSUE THREAD

**Issue by Superpat** [yesodweb/yesod#1313]
When I try running my app on a fresh install of freebsd-11, I cant get past the refresh page. It doesnt matter if I use ``stack exec app`` or ``stack exec yesod devel``, I get the same error: ``devel.hs: getAddrInfo: does not exist (hostname nor servname provided, or not known)``

Another issue mentions this might be related to there not being a v4 address, but the output of ``ifconfig`` tells me there is one.

``
ale0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
        options=c319a<TXCSUM,VLAN_MTU,VLAN_HWTAGGING,VLAN_HWCSUM,TSO4,WOL_MCAST,WOL_MAGIC,VLAN_HWTSO,LINKSTATE>
        ether e0:cb:4e:0e:17:70
        inet6 fe80::e2cb:4eff:fe0e:1770%ale0 prefixlen 64 scopeid 0x1
        inet 192.168.2.7 netmask 0xffffff00 broadcast 192.168.2.255
        nd6 options=23<PERFORMNUD,ACCEPT_RTADV,AUTO_LINKLOCAL>
        media: Ethernet autoselect (100baseTX <full-duplex>)
        status: active
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
        options=600003<RXCSUM,TXCSUM,RXCSUM_IPV6,TXCSUM_IPV6>
        inet6 ::1 prefixlen 128
        inet6 fe80::1%lo0 prefixlen 64 scopeid 0x2
        inet 127.0.0.1 netmask 0xff000000
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        groups: lo
``

Very possibly related to #1285

**Comment by snoyberg** [comment]
Can you try to reproduce with a minimal WAI applications? Even just:

```haskell
#!/usr/bin/env stack
-- stack --resolver lts-6.23 runghc --package warp
import Network.Wai.Handler.Warp

main = run 3000 undefined
```

**Comment by Superpat** [comment]
Running that and going on localhost:3000 gives me a 500 error with the following text:

Something went wrong

**Comment by snoyberg** [comment]
Cool. How about the following?

```haskell
#!/usr/bin/env stack
-- stack --resolver lts-6.23 runghc --package warp
{-# LANGUAGE OverloadedStrings #-}
import Network.Wai.Handler.Warp

main = runSettings (setHost "*4" defaultSettings) undefined
```

And can you play around with various host settings (like `"*6"`, `"127.0.0.1"`) and see what fails and what doesn't?

**Comment by Superpat** [comment]
The new script works the same, I tried ``*"6", "127.0.0.1"`` and those worked fine on the script, but did not work with the yesod app, I also tried inputting values from ``ifconfig`` ``"192.168.2.7", "192.168.2.255.", "fe80::e2cb:4eff:fe0e:1770"`` but, predictably, they did not work.

**Comment by snoyberg** [comment]
The default host for your application in `stack exec app` should be `*4`, so I'm not sure what could be causing the difference. Can you try running the script above with the same resolver setting as your app to eliminate any other possible difference? Also, perhaps try compiling and running the script instead of using `runghc`?

Also, can you clarify exactly what error message you get from the other IP addresses you tried out?

**Comment by Superpat** [comment]
I am running the same resolver in both, 6.23

The error message was the same as with the valid ones, in the case of the yesod app. The script however does not give an error, but firefox reports that there is nothing on localhost:3000.

How do I quickly compile the script? Do I have to start a stack project?

**Comment by snoyberg** [comment]
No, you can use stack ghc, with the same arguments as runghc is receiving
in the top of that file.

**Comment by Superpat** [comment]
Building the script with stack ghc changes nothing, it still works the same.

**Comment by snoyberg** [comment]
I'm sorry, I'm out of ideas. I don't know what would be different in your application/the scaffolding versus the minimal examples here, and unfortunately have no way of reproducing this problem myself. Perhaps reach out on a mailing list and see if other FreeBSD users have encountered this?

**Comment by Superpat** [comment]
Thanks for the help anyways, the only difference I could think of was my
pf config, but deactivating pf didnt change anything. I 'll se on the
freebsd mailing lists.


--

  Patrick Marchand

www.patrickmarchand.com

**Comment by MaxGabriel** [comment]
Since this issue was created, I added a patch to network to print a more verbose error message. This should give you the exact arguments to getAddrInfo if you want to jump straight to debugging from that level.
